src/shared/FeatureFlag/FeatureFlag.tsx

```diff
@@ -1,15 +1,70 @@
-import React from 'react'
+import React, { FC, PropsWithChildren, ReactNode } from 'react'
 
 import { useFeatureFlag } from 'libs/firebase/firestore/featureFlags/useFeatureFlag'
 import { RemoteStoreFeatureFlags } from 'libs/firebase/firestore/types'
 
 type FeatureFlagProps = {
-  featureFlag: RemoteStoreFeatureFlags
-  children: React.JSX.Element
+  featureFlag: keyof typeof RemoteStoreFeatureFlags
+  children: ReactNode
 }
 
-export function FeatureFlag({ featureFlag, children }: FeatureFlagProps) {
-  const featureFlagEnabled = useFeatureFlag(featureFlag)
+type SubComponentProps = PropsWithChildren
 
-  return featureFlagEnabled ? children : null
+/**
+ * FeatureFlag component to conditionally render children based on the feature flag state.
+ *
+ * @param {keyof RemoteStoreFeatureFlags} featureFlag - The name of the feature flag to check.
+ * @param {ReactNode} children - The children components, typically `FeatureFlag.On` and `FeatureFlag.Off`.
+ * @example
+ * <FeatureFlag featureFlag={RemoteStoreFeatureFlags.WIP_APP_V2_CATEGORY_BLOCK}>
```

Comment:
il me semble que le typage impose une des clés de l'enum

```suggestion
 * <FeatureFlag featureFlag="WIP_APP_V2_CATEGORY_BLOCK">
```

Reply:
Effectivement, pas besoin d'import de RemoteStoreFeatureFlag et le resultat encapsulé est le même.
Si une clé change, le Typescript le fera remonter.
Le développeur à simplement à ouvrir les guillemets et obtiens une autocomplétion, bien plus pratique à mon sens